shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -34,6 +34,8 @@
 
 static constexpr int kMicrosecondsPerSecond = 1000 * 1000;
 static constexpr CGFloat kScrollViewContentSize = 2.0;
+static constexpr fml::TimeDelta kKeyboardAnimationUpdateViewportMetricsDelay =
```

Comment:
This does make the animation better, but an arbitrary delay could also cause janks. 

Maybe instead of relying on a arbitrary delay, we could try a solution where the keyboard animation and rasterizer is synchronized, similar to macOS PlatformView.
One thing to try is to get a callback from the Shell when frame is finished rasterizing and synchronously run the keyboard animation with a mutex.
https://github.com/flutter/engine/blob/main/shell/common/shell.cc#L1447

Reply:
Since an arbitrary delay maybe cause some another issue. What I m going to do is to make `SetViewportMetrics` call after vsync process callback. So I think maybe using raster callback is a little bit late. I will raise a new PR for changing that. This marks as draft. ^_^